nntrainer/graph/graph_core.cpp

```diff
@@ -59,27 +45,47 @@ GraphCore::getSortedNode(unsigned int ith) const {
   return Sorted[ith];
 }
 
+void GraphCore::makeAdjacencyList(
+  std::vector<std::list<std::shared_ptr<GraphNode>>> &adj) {
+  /** initialize the adj list */
+  for (auto &node : node_list) {
+    adj.push_back(std::list<std::shared_ptr<GraphNode>>({node}));
+  }
+
+  /** make the connections */
+  for (auto &node : node_list) {
+    for (auto const &in_conn : node->getInputConnections()) {
+      if (istrequal(in_conn, "__data__"))
+        continue;
+      unsigned int to_node_id = getNode(in_conn)->getIndex();
+      adj[to_node_id].push_back(node);
+    }
+  }
+}
+
 void GraphCore::topologicalSortUtil(
```

Comment:
How about making this as a static free function inside this source file?

Reply:
will do.

Reply:
actually, this cant be done. topologicalSortUtil uses getNode() which requires it to be member function.
There is work around to use directly get it from adj, but it uses the knowledge of how index is assigned, and I dont think its a good idea.

Reply:
Okay